### ISSUE TITLE
Incorrect union narrowing in chained conditionals

### ISSUE THREAD

**Issue by JaviFuentes94** [python/mypy#12208]
**Bug Report**
Hey there! 

I have recently encountered this case: 
```
class Animal:
    name: Literal["animal"]
    n_legs: int

class Plant:
    name: Literal["plant"]

def get_n_legs(item: Union[Animal, Plant]):
    if item.name == "animal" and item.n_legs > 6:    # This fails mypy with union-attr for n_legs in Plant
        return "Many" 
```

**Expected Behavior**

Mypy should be able to narrow down chained conditionals as presented. It seems that pylance has no problems with it, for example.

**Comment by JelleZijlstra** [comment]
Doesn't repro for me either: https://mypy-play.net/?mypy=latest&python=3.10&gist=e2c735beb847d6da4d53477050f3a119

**Comment by mikaelarguedas** [comment]
Hey there,

I believe I ran into another occurrence of this. It looks like the type has not been narrowed by the first if statement.
Happens with lates mypy on Python3.8.
Trying it on https://mypy-play.net  looks to happen on more recent versions of Python as well.

Reproducible example: https://mypy-play.net/?mypy=latest&python=3.11&flags=verbose&gist=07d004a22a748059c834ff42d690c2ed

```python
from typing import Optional, Union


class RangeVals:
    """Hold a parameter value range."""

    def __init__(
        self,
        min_val: Union[int, float],
        max_val: Union[int, float],
    ) -> None:
        self.min_val = min_val
        self.max_val = max_val


def functionn(default_val: Union[str, float, int], rangeval: RangeVals):
    if type(default_val) != type(rangeval.min_val):
        raise TypeError("foo")
    if default_val < rangeval.min_val:
        raise ValueError("bar")


functionn("abc", RangeVals(min_val=3, max_val=10))
```

**Comment by erictraut** [comment]
The original bug in this issue appears to have been fixed in mypy 1.4.

@mikaelarguedas, your issue is unrelated to OP's bug. The behavior in your sample is expected behavior in mypy.

**Comment by hauntsaninja** [comment]
I fixed this in #11521
